spec/formatting.md

```diff
@@ -134,7 +139,7 @@ Finally, sort the list of variants `vars` and select the _pattern_:
 The method SortVariants is determined by the implementation.
 It takes as an argument a `sortable` list of (integer, _variant_) tuples,
 which it modifies in place using some stable sorting algorithm.
-The method does not return anything.
+The method does not return a value: `sortable` is its out-parameter.
```

Comment:
"out-parameter"?

I actually find this a bit odd or maybe constraining. This defines a method whose output is a side-effect on the input value, which is unusual and it forbids writing a method with a signature like:

```java
public MessageKeyList sortVariants(final MessageKeyList input)
```

Why is it necessary for us to force this?

Reply:
This is intended as a convenience for keeping the method simple. Recall that the section includes this near the top:
> An implementation MAY use any pattern selection method, as long as its observable behaviour matches the results of the method defined here.

On the other hand, if the sense is that it would be easier to read if the implementation-defined method returned some value that was then re-assigned to `sortable`, that's certainly a possible change.

Reply:
@aphillips The way the algorithm is currently written, `sortable` has to change on each iteration of the `while` loop. However, I think I've found a way to leave it open whether `SortVariants` modifies the list in-place and returns the same list as its result, or returns a new list. Let me know what you think.